src/mpl/src/gavl/mpl_gavl.c

```diff
@@ -333,117 +568,77 @@ int MPL_gavl_tree_free(MPL_gavl_tree_t gavl_tree)
 int MPL_gavl_tree_delete(MPL_gavl_tree_t gavl_tree, const void *addr, uintptr_t len)
 {
     int mpl_err = MPL_SUCCESS;
+    void *val;
     gavl_tree_node_s *cur_node;
-    gavl_tree_node_s *inorder_node;
+    gavl_tree_subnode_s *cur_sub, *prev_sub;
     gavl_tree_s *gavl_tree_iptr = (gavl_tree_s *) gavl_tree;
 
-    DECLARE_STACK(gavl_tree_node_s *, node_stack);
-
-    void *val;
     do {
-        CLEAR_STACK(node_stack);
         cur_node = gavl_tree_iptr->root;
-        val = NULL;
         if (cur_node == NULL) {
             goto fn_exit;
         } else {
+            val = NULL;
             do {
                 int cmp_ret = gavl_intersect_cmp_func((uintptr_t) addr, len, cur_node);
```

Comment:
Why the internal search logic in `MPL_gavl_tree_delete ` is different from that in `MPL_gavl_tree_insert`? Should both of them perform the same search logic before the consequent `delete` or `insert`?

Reply:
no, they are very different.
`MPL_gavl_tree_delete` will delete all subnodes that intersect input key (addr, len) 
`MPL_gavl_tree_insert` will merge all tree nodes that intersect input key (addr, len)
Their algorithms are different, so logic is different
(please see my comments for `MPL_gavl_tree_delete` and `MPL_gavl_tree_insert`)